packages/fetchai/skills/registration_aw1/handlers.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains a scaffold of a handler."""
+
+from typing import Optional, cast
+
+from aea.protocols.base import Message
+from aea.skills.base import Handler
+
+from packages.fetchai.protocols.register.message import RegisterMessage
+from packages.fetchai.protocols.signing.message import SigningMessage
+from packages.fetchai.skills.registration_aw1.dialogues import (
+    RegisterDialogue,
+    RegisterDialogues,
+    SigningDialogue,
+    SigningDialogues,
+)
+from packages.fetchai.skills.registration_aw1.strategy import Strategy
+
+
+class AW1RegistrationHandler(Handler):
+    """This class handles register messages."""
+
+    SUPPORTED_PROTOCOL = RegisterMessage.protocol_id
+
+    def setup(self) -> None:
+        """
+        Implement the setup.
+
+        :return: None
+        """
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to an envelope.
+
+        :param message: the message
+        :return: None
+        """
+        register_msg = cast(RegisterMessage, message)
+
+        # recover dialogue
+        register_dialogues = cast(RegisterDialogues, self.context.register_dialogues)
+        register_dialogue = cast(
+            Optional[RegisterDialogue], register_dialogues.update(register_msg)
+        )
+        if register_dialogue is None:
+            self._handle_unidentified_dialogue(register_msg)
```

Comment:
Quality of life change:
Missing some logging info here and everywhere else 

(but not that high priority to get things going)

Reply:
yep can be done but as you say, low priority atm